common/src/main/java/io/homeassistant/companion/android/common/util/AudioUrlPlayer.kt

```diff
@@ -33,13 +34,20 @@ class AudioUrlPlayer(private val audioManager: AudioManager?) {
      * Stream and play audio from the provided [url]. Any currently playing audio will be stopped.
      * This function will suspend until playback has started.
      * @param isAssistant whether the usage/stream should be set to Assistant on supported versions
+     * @param donePlaying callback to be invoked when playback has finished (it covers when it's not playing anything or an error happened)
      * @return `true` if the audio playback started, or `false` if not
      */
-    suspend fun playAudio(url: String, isAssistant: Boolean = true): Boolean = withContext(Dispatchers.IO) {
+    suspend fun playAudio(url: String, isAssistant: Boolean = true, donePlaying: () -> Unit): Boolean = withContext(Dispatchers.IO) {
         if (player != null) {
             stop()
         }
 
+        val refDonePlaying = AtomicReference<(() -> Unit)?>(donePlaying)
```

Comment:
These types don't match - `donePlaying` can be null here but not in the function arguments.

Reply:
Yes that's on purpose. I nullify the reference once the player is done with it. I do that especially because of the error case. It is just for safety if the API in the future changes I don't want this callback to be ever called twice for any reason.

I can add a comment if it's not clear enough.

Reply:
Would it make sense to also allow `null` as an argument then? No reason to pass a function that does nothing.

Reply:
Indeed I'll make it possible to set this as null. I didn't because it is used in only one place at the moment, but you are right if it used in another place it might not be needed.